src/main/java/jenkins/plugins/office365connector/Webhook.java

```diff
@@ -51,12 +56,31 @@
     private List<Macro> macros = Collections.emptyList();
 
     @DataBoundConstructor
-    public Webhook(String url) {
-        this.url = url;
+    public Webhook() {
+    }
+
+    public String getUrl(Run run) {
+        if (this.url != null) {
```

Comment:
not very good about strings - how about "" ?

Reply:
`url` and `urlCredentialsId` are always set using `fixEmptyAndTrim` so `""` is normalized to `null`.

https://github.com/jenkinsci/office-365-connector-plugin/blob/2560636e8c2ff8e56a7a27beadbd8823084fd883/src/main/java/jenkins/plugins/office365connector/Webhook.java#L77-L79